### PR TITLE
Add support for autoscaling v2

### PR DESCRIPTION
**Pull request by tuananhnguyen-ct**
Signed-off-by: Tuan Anh Nguyen <tuananh.nguyen-ext@commercetools.de>

Fixes https://github.com/vectordotdev/helm-charts/issues/186

### CODE REVIEW

File: charts/vector/templates/haproxy/_helpers.tpl [hunk 1]
```diff
@@ -38,3 +38,14 @@ Create the name of the service account to use
...
 {{- end }}
+
+{{/*
+Return the appropriate apiVersion for HPA autoscaling APIs.
+*/}}
+{{- define "autoscaling.apiVersion" -}}
+{{- if or (.Capabilities.APIVersions.Has "autoscaling/v2") (semverCompare ">=1.23" .Capabilities.KubeVersion.Version) -}}
+"autoscaling/v2"
+{{- else -}}
+"autoscaling/v2beta2"
```
Comment (spencergilbert) [inline charts/vector/templates/haproxy/_helpers.tpl:49]: I think we'll want to keep a check for `v2beta2` support to not cause issues with clusters that are so old that they don't even support that version. Though I'm not seeing how far back that would end up being.

File: charts/vector/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: vector
-version: "0.14.0"
+version: "0.14.1"
```
Comment (spencergilbert) [inline charts/vector/Chart.yaml:3]: I'd classify this as a new feature/enhancement and bump it to `0.15`

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: charts/vector/templates/haproxy/_helpers.tpl (hunk 1 above)
Comment (tuananhnguyen-ct) [reply charts/vector/templates/haproxy/_helpers.tpl:49]: v2beta2 is available from k8s 1.12, and it'll require some changes in HPA config (targetAverageUtilization vs averageUtilization) if we go back to v2beta1, while vector only support k8s 1.15+. I don't see why we really need to go back for v2beta1 support for now.

Comment (spencergilbert) [reply charts/vector/templates/haproxy/_helpers.tpl:49]: Thanks - that's definitely far back enough!

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**